Source/Core/Core/HW/SI/SI_DeviceGBA.cpp

```diff
@@ -330,13 +330,15 @@ int CSIDevice_GBA::RunBuffer(u8* buffer, int request_length)
   case NextAction::ReceiveResponse:
   {
     int num_data_received = m_sock_server.Receive(buffer);
-    m_next_action = NextAction::SendCommand;
-    if (num_data_received == 0)
+    if (num_data_received < 0)
     {
+      m_next_action = NextAction::SendCommand;
       u32 reply = Common::swap32(SI_ERROR_NO_RESPONSE);
       std::memcpy(buffer, &reply, sizeof(reply));
       return sizeof(reply);
     }
+    if (num_data_received == 0)
```

Comment:
Nit: If this `if` is done first, since it's a return, you don't need to move (or duplicate) the `m_next_action` assignment; it can be done between the first `if`'s return and subsequent checks.

Reply:
Thank you, I changed it and added a comment.